internal/store/atlas/backup.go

```diff
@@ -17,39 +17,45 @@ package atlas
-
+//go:generate mockgen -destination=../../mocks/atlas/mock_backup.go -package=atlas github.com/mongodb/mongodb-atlas-cli/internal/store/atlas CompliancePolicyDescriber,CompliancePolicy,CompliancePolicyEncryptionAtRestUpdater
 type CompliancePolicyDescriber interface {
 	DescribeCompliancePolicy(projectID string) (*atlasv2.DataProtectionSettings, error)
 }
+type CompliancePolicyUpdater interface {
+	UpdateCompliancePolicy(projectID string, opts *atlasv2.DataProtectionSettings) (*atlasv2.DataProtectionSettings, error)
+}
+type CompliancePolicyEncryptionAtRestUpdater interface {
+	UpdateEncryptionAtRest(projectID string, enable bool) (*atlasv2.DataProtectionSettings, error)
+}
 
 type CompliancePolicy interface {
-	DescribeCompliancePolicy(projectID string) (*atlasv2.DataProtectionSettings, error)
-	UpdateCompliancePolicy(projectID string, opts *atlasv2.DataProtectionSettings) (*atlasv2.DataProtectionSettings, error)
+	CompliancePolicyDescriber
+	CompliancePolicyUpdater
 }
 
-// DescribeCompliancePolicy encapsulates the logic to manage different cloud providers.
 func (s *Store) DescribeCompliancePolicy(projectID string) (*atlasv2.DataProtectionSettings, error) {
-	switch s.service {
-	case config.CloudService, config.CloudGovService:
-		result, _, err := s.clientv2.CloudBackupsApi.GetDataProtectionSettings(s.ctx, projectID).Execute()
-		return result, err
-	default:
-		return nil, fmt.Errorf("%w: %s", errUnsupportedService, s.service)
-	}
+	result, _, err := s.clientv2.CloudBackupsApi.GetDataProtectionSettings(s.ctx, projectID).Execute()
+	return result, err
 }
 
-// UpdateCompliancePolicy encapsulates the logic to manage different cloud providers.
 func (s *Store) UpdateCompliancePolicy(projectID string, opts *atlasv2.DataProtectionSettings) (*atlasv2.DataProtectionSettings, error) {
-	switch s.service {
-	case config.CloudService, config.CloudGovService:
-		result, _, err := s.clientv2.CloudBackupsApi.UpdateDataProtectionSettings(s.ctx, projectID, opts).Execute()
-		return result, err
-	default:
-		return nil, fmt.Errorf("%w: %s", errUnsupportedService, s.service)
+	result, _, err := s.clientv2.CloudBackupsApi.UpdateDataProtectionSettings(s.ctx, projectID, opts).Execute()
+	return result, err
+}
+
+func (s *Store) UpdateEncryptionAtRest(projectID string, enable bool) (*atlasv2.DataProtectionSettings, error) {
+	compliancePolicy, _, err := s.clientv2.CloudBackupsApi.GetDataProtectionSettings(s.ctx, projectID).Execute()
+	if err != nil {
+		return nil, fmt.Errorf("couldn't update encryption at rest: %w", err)
+	}
+	compliancePolicy.SetEncryptionAtRestEnabled(enable)
+
+	result, _, err := s.clientv2.CloudBackupsApi.UpdateDataProtectionSettings(s.ctx, projectID, compliancePolicy).Execute()
+	if err != nil {
+		return nil, fmt.Errorf("couldn't update encryption at rest: %w", err)
 	}
+	return result, nil
 }
```

Comment:
This code is taken from https://github.com/mongodb/mongodb-atlas-cli/pull/2168 <- please ignore here!